src/main/drivers/system.c

```diff
@@ -79,7 +88,19 @@ uint32_t micros(void)
          */
         asm volatile("\tnop\n");
     } while (ms != sysTickUptime);
-    return (ms * 1000) + (usTicks * 1000 - cycle_cnt) / usTicks;
+
+    // SysTick interrupts never get delivered above if called from ISRs.
+    // Treat COUNTFLAG like a pending interrupt flag.
+
+    if (SysTick->CTRL & SysTick_CTRL_COUNTFLAG_Msk) {
```

Comment:
Race (one of):
Systick expired, SysTick_CTRL_COUNTFLAG_Msk set
IrqLo: CTRL read (and cleared)
IrqHi: CTRL read, flag not set - invalid time
IrqLo: sysTickPending = 1; time ok
Systick: sysTickUptime++

Even worse, it is possible to read CTRL just before SysTick_Handler is executed, so that bit is set but ISR executes before setting sysTickPending - it will be set until next ISR ... 

Using SYSTICKACT will help with part of this problem. Writing separate interruptible and non-interuptibe version will help a lot ...